internal/app/secretless/handlers/mysql/backend.go

```diff
@@ -76,18 +120,20 @@ func (h *Handler) ConnectToBackend() (err error) {
 
 	address := h.BackendConfig.Host + ":" + strconv.FormatUint(uint64(h.BackendConfig.Port), 10)
 
-	if backend, err = net.Dial("tcp", address); err != nil {
+	h.BackendConnection, err = net.Dial("tcp", address)
+	if err != nil {
```

Comment:
There are like 2 million of these checks that just return when not nil in this file.  Here's a trick I saw Rob doing that will probably get rid of 20+ lines of code and clean it up:

```go
	check := func() {
		if err != nil {
			t.Fatal(err)
		}
	}
```
Then replace all the current checks with:
```go
check()
```

Reply:
`t.Fatal(err)` will panic. these are errors that need not result in panic, they need to be explicitly returned

Reply:
yep you are 100% right, i read that too fast.